### PR TITLE
Add gsoc project for fox

### PR DESCRIPTION
**Pull request by jvican**
I have another project that I want to submit and I will towards the end of the
day.

### CODE REVIEW

**Comment by jvican** [comment]
Review by @olafurpg.

File: gsoc/2018.md [hunk 1]
```diff
@@ -81,6 +81,37 @@ Scalafix is a fast moving project, so you will be working in a codebase with mul
...
+
+The scope of the project is to continue the development of an existing
+prototype, called `fox`. The project can start small with simple changes to get
+familiar with the codebase, and gradually step up as progress happens.
+
+Your end goal will be to implement the following features:
+
+1. A [Flexmark][] integration to generate PDF and ebooks (epub, mobi) using
+   Pandoc.
+1. A series of modifications to the default website template that:
```
Comment (olafurpg) [inline gsoc/2018.md]: Another nice-to-have feature I'd love in a doctool: local server that refreshes browser on edit, like https://www.gatsbyjs.org/, http://www.mkdocs.org/ and https://gohugo.io/ have

File: gsoc/2018.md [hunk 2]
```diff
@@ -81,6 +81,37 @@ Scalafix is a fast moving project, so you will be working in a codebase with mul
...
+### Fox, a documentation engine for Scala
+
+Documentation may be consumed and written in many ways. Library authors can
+write tutorials, in-depth guides, or Scala docs; and users can consume them in
+PDF files, online websites or ebook readers. There are many hidden requirements
+in writing good documentation and, given how difficult it is to get it right,
+it is a task that many developers eventually ignore or, at best, procrastinate
+at.
+
+This project calls for the creation of a flexible documentation engine that
```
Comment (olafurpg) [inline gsoc/2018.md]: I think it's good to mention that there already exist a lot of fantastic doc tools like https://www.gatsbyjs.org/, http://www.mkdocs.org/ and https://gohugo.io/ . However, none of them integrate nicely with the Scala toolchain like sbt/tut.

File: gsoc/2018.md [hunk 3]
```diff
@@ -81,6 +81,37 @@ Scalafix is a fast moving project, so you will be working in a codebase with mul
...
+1. A series of modifications to the default website template that:
+  1. Implement reasonable UI standards for most projects.
+  1. Implement good search and discovery capabilities.
+
+You need to be familiar with Scala and enjoy building tools for developers.
+
+Supervised by [@jvican](https://github.com/jvican).
+
+[Flexmark]: https://github.com/vsch/flexmark-java/
+[Semanticdb]: https://github.com/scalameta/scalameta
```
Comment (olafurpg) [inline gsoc/2018.md]: Unused link

File: gsoc/2018.md [hunk 4]
```diff
@@ -81,6 +81,37 @@ Scalafix is a fast moving project, so you will be working in a codebase with mul
...
+
+Your end goal will be to implement the following features:
+
+1. A [Flexmark][] integration to generate PDF and ebooks (epub, mobi) using
+   Pandoc.
+1. A series of modifications to the default website template that:
+  1. Implement reasonable UI standards for most projects.
+  1. Implement good search and discovery capabilities.
+
+You need to be familiar with Scala and enjoy building tools for developers.
```
Comment (olafurpg) [inline gsoc/2018.md]: I would say "Experience with HTML/CSS is required", this is in fact a great project to learn Scala!

**Comment by smarter** [comment]
I think it'd be nice if we could see how this work could be integrated with [Dottydoc](https://github.com/lampepfl/dotty/tree/master/doc-tool):
- It already uses flexmark
- It also uses [liqp](https://github.com/bkiers/Liqp) to provide a templating language that looks like Jekyll, allowing you to generate any kind of static site you want. Every page of http://dotty.epfl.ch/ is generated that way. The sources for the templates are in https://github.com/lampepfl/dotty/tree/master/doc-tool/resources

Currently, Dottydoc relies on the Dotty compiler to generate API documentation, but the plan is to store the API doc in Tasty instead (@Duhemm already started work on this) so that Dottydoc will just take Tasty as input (This should also make it possible to get incremental documentation generation in Dottydoc, and make it really easy to display the documentation in other tools such as an IDE or a REPL). Dottydoc only supports outputing HTML but it'd be interesting to support other formats such as PDF indeed.

**Comment by SethTisue** [comment]
there is a lot of existing art in this area. for this to be a convincing proposal, I'd need to see some sort of justification why existing tools are 1) inadequate, and 2) cannot be improved to be adequate, necessitating starting over

I don't even know this space that well, but just off the top of my head:

* https://github.com/lightbend/paradox
* http://www.foundweekends.org/pamflet/
* https://github.com/szeiger/ornate
* whatever @lihaoyi uses to do his doc sites such as http://ammonite.io

perhaps I misunderstand, and your goals are actually substantially different than the goals of these projects? I don't know.

**Comment by SethTisue** [comment]
also, who will maintain this after the GSoC student finishes it? the Scala Center? do we need an advisory board proposal for that?

**Comment by SethTisue** [comment]
that helps, thx

**Comment by smarter** [comment]
> Putting that in the scope of this project is too much for an student to do in three months. That should be an independent project.

But if you start from dottydoc, there's a lot of things that you wouldn't need to reimplement since they're already implemented I think.

**Comment by jvican** [comment]
> But if you start from dottydoc, there's a lot of things that you wouldn't need to reimplement since they're already implemented I think.
  
I'm pretty sure we'll be able to draw inspiration from DottyDoc, but my primary goal to mentor this project is to have something that works for Scala 2.x because I'd like to use it in my projects. Thanks for the reference to [liqp](https://github.com/bkiers/Liqp), it may prove useful later down the road.

**Comment by MasseGuillaume** [comment]
@jonas Would you like to co-mentor this project?

File: gsoc/2018.md [hunk 5]
```diff
@@ -81,6 +81,35 @@ Scalafix is a fast moving project, so you will be working in a codebase with mul
...
+in writing good documentation and, given how difficult it is to get it right,
+it is a task that many developers eventually ignore or, at best, procrastinate
+at.
+
+This project calls for the creation of a flexible documentation engine that
+aims to simplify writing documentation for Scala library authors, so that they
+can focus on the inherent difficulty of writing good docs.
+
+The scope of the project is to continue the development of an existing
+prototype. The project can start small with simple changes to get familiar with
```
Comment (julienrf) [inline gsoc/2018.md]: You should add a link to the prototype

Comment (jvican) [reply gsoc/2018.md]: The prototype is not yet offline

**Comment by julienrf** [comment]
Oh, I didn’t see the existing discussion. I see now that my questions overlap with Seth’s questions, and have been answered in the comments.

**Comment by jvican** [comment]
I've just realised I have too much in my plate and won't be able to properly mentor this project.

**Pull request closed without merging**